lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -3362,8 +3362,19 @@ static void rewriteFunction(AddressLoweringState &pass) {
     if (valueDef->getType().isAddress())
       continue;
 
+    SmallPtrSet<Operand *, 8> originalUses;
     SmallVector<Operand *, 8> uses(valueDef->getUses());
-    for (Operand *oper : uses) {
+    for (auto *oper : uses) {
+      originalUses.insert(oper);
+      UseRewriter::rewriteUse(oper, pass);
+    }
+    uses.clear();
+    for (auto *use : valueDef->getUses()) {
+      uses.push_back(use);
```

Comment:
I would check membership in originalUses before inserting them in a worklist.

Reply:
Fixed.